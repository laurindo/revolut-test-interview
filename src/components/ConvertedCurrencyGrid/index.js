import React from 'react';
import PropTypes from 'prop-types';
import SimpleGridList from '../SimpleGridList';
import Loading from '../Loading';

import './style.css';

const ConvertedCurrencyGrid = ({ props, currencyConversion, valuesTitle }) => (
  <div className='container'>
    <div className="row">
      {
        (props.loading) ?
          <Loading /> :
          <SimpleGridList
            {...props}
            values={ props.convertedValues }
            valuesTitle={ valuesTitle }
            valuesRows={ currencyConversion() }
          />
      }
    </div>
  </div>
);

ConvertedCurrencyGrid.propTypes = {
  props: PropTypes.object.isRequired,
  currencyConversion: PropTypes.func.isRequired,
  valuesTitle: PropTypes.array.isRequired,
};

export default ConvertedCurrencyGrid;
