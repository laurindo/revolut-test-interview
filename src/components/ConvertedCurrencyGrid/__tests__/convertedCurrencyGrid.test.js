import React from 'react';
import { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ConvertedCurrencyGrid from '../index';

configure({ adapter: new Adapter() });

describe('Converted CUrrency Grid Component', () => {
  test('should render the component', () => {
    const props= {
      loading: false,
    };

    const component = shallow(
      <ConvertedCurrencyGrid
        valuesTitle={['Currency', 'Unit Price', 'Balance Total']}
        props={ props }
        currencyConversion={ () => ['USD', '$ 1.00', '$ 1.34'] } />
    );

    expect(component).toMatchSnapshot();
    expect(component.find('.row')).toBeDefined();
    expect(component.find('.row').get(0).props.className).toEqual('row');
    expect(component.find('.row').get(0).props.children.props.loading).toEqual(false);
    expect(component.find('.row').get(0).props.children.props.valuesTitle[0]).toEqual('Currency');
    expect(component.find('.row').get(0).props.children.props.valuesTitle[1]).toEqual('Unit Price');
    expect(component.find('.row').get(0).props.children.props.valuesTitle[2]).toEqual('Balance Total');

    expect(component.find('.row').get(0).props.children.props.valuesRows[0]).toEqual('USD');
    expect(component.find('.row').get(0).props.children.props.valuesRows[1]).toEqual('$ 1.00');
    expect(component.find('.row').get(0).props.children.props.valuesRows[2]).toEqual('$ 1.34');
  });
});
