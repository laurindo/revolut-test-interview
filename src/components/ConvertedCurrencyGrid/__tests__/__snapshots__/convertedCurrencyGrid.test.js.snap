exports[`Converted CUrrency Grid Component should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConvertedCurrencyGrid
    currencyConversion={[Function]}
    props={
        Object {
            "loading": false,
          }
    }
    valuesTitle={
        Array [
            "Currency",
            "Unit Price",
            "Balance Total",
          ]
    } />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row">
        <SimpleGridList
                loading={false}
                values={undefined}
                valuesRows={
                        Array [
                                "USD",
                                "$ 1.00",
                                "$ 1.34",
                              ]
                }
                valuesTitle={
                        Array [
                                "Currency",
                                "Unit Price",
                                "Balance Total",
                              ]
                } />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <SimpleGridList
          loading={false}
          values={undefined}
          valuesRows={
                    Array [
                              "USD",
                              "$ 1.00",
                              "$ 1.34",
                            ]
          }
          valuesTitle={
                    Array [
                              "Currency",
                              "Unit Price",
                              "Balance Total",
                            ]
          } />,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "loading": false,
          "values": undefined,
          "valuesRows": Array [
            "USD",
            "$ 1.00",
            "$ 1.34",
          ],
          "valuesTitle": Array [
            "Currency",
            "Unit Price",
            "Balance Total",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row">
          <SimpleGridList
                    loading={false}
                    values={undefined}
                    valuesRows={
                              Array [
                                        "USD",
                                        "$ 1.00",
                                        "$ 1.34",
                                      ]
                    }
                    valuesTitle={
                              Array [
                                        "Currency",
                                        "Unit Price",
                                        "Balance Total",
                                      ]
                    } />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <SimpleGridList
            loading={false}
            values={undefined}
            valuesRows={
                        Array [
                                    "USD",
                                    "$ 1.00",
                                    "$ 1.34",
                                  ]
            }
            valuesTitle={
                        Array [
                                    "Currency",
                                    "Unit Price",
                                    "Balance Total",
                                  ]
            } />,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "loading": false,
            "values": undefined,
            "valuesRows": Array [
              "USD",
              "$ 1.00",
              "$ 1.34",
            ],
            "valuesTitle": Array [
              "Currency",
              "Unit Price",
              "Balance Total",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
